cns/restserver/ipam.go

```diff
@@ -116,6 +116,23 @@ func (service *HTTPRestService) RequestIPConfigHandler(w http.ResponseWriter, r
 		return
 	}
 
+	// Check the status of the NC
+	ncStatuses := multiContainerStatus(service.state.ContainerStatus)
+	if unsuccessfulStatuses := ncStatuses.GetUnsuccessfulStatusErrors(); unsuccessfulStatuses != "" {
+		// If the status is anything other than success, we send a response back with the actual status and NC ID.
+		reserveResp := &cns.IPConfigResponse{
+			Response: cns.Response{
+				ReturnCode: types.FailedToAllocateIPConfig,
+				Message:    unsuccessfulStatuses,
+			},
+		}
+
+		w.Header().Set(cnsReturnCode, reserveResp.Response.ReturnCode.String())
+		err = service.Listener.Encode(w, &reserveResp)
+		logger.ResponseEx(service.Name+operationName, ipconfigRequest, reserveResp, reserveResp.Response.ReturnCode, err)
+		return
+	}
```

Comment:
By checking the NCStatus early you are blocking the call to the service. I don't think it is the right behavior here. This will block IP allocations from the IPs that are available on the node itself and are getting reclaimed while the NC status is SubnetFull or something else. We should not be determining the response state from the state of the NC. Basically it should be more of a helper method to add to the response message on line 153.